app/components/projects/delete_list_modal_component.html.erb

```diff
@@ -1,25 +1,14 @@
-<%= render(
-      Primer::Alpha::Dialog.new(
-        title: t(:"projects.lists.delete_modal.title"),
-        size: :large,
-        id: MODAL_ID,
-        data: { "test-selector": MODAL_ID }
-      )
-    ) do |d| %>
-  <% d.with_header(variant: :large) %>
-  <% d.with_body { t(:"projects.lists.delete_modal.text") } %>
-  <% d.with_footer do %>
-    <%= render(Primer::Beta::Button.new(data: { "close-dialog-id": MODAL_ID })) { I18n.t(:button_cancel) } %>
-    <%= form_with(
-          url: project_query_path(query),
-          method: :delete
-        ) do %>
-      <%= render(
-            Primer::Beta::Button.new(
-              scheme: :danger,
-              type: :submit
-            )
-          ) { I18n.t(:button_delete) } %>
-    <% end %>
-  <% end %>
-<% end %>
+<%=
+  render(Primer::OpenProject::DangerDialog.new(title: t(:"projects.lists.delete_modal.title"),
+                                               form_arguments: {
+                                                 action: project_query_path(query),
+                                                 method: :delete
+                                               },
+                                               id: MODAL_ID,
+                                               test_selector: MODAL_ID)) do |d|
+    d.with_confirmation_message do |message|
+      message.with_heading(tag: :h2) { t(:"projects.lists.delete_modal.title") }
```

Comment:
@HDinger /cc @psatyal Sorry for the late feedback.

It would be good if we could adopt the same style here - an active question (and also use the same I18n key naming convention`)

**Delete this project list?**

rather than

**Delete project list** (this is OK as an accessibility title though)

WDYT?